integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/adapters.py

```diff
@@ -108,16 +112,30 @@ def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
             - `prompt`: The prompt to be sent to the model.
             - specified inference parameters.
         """
-        default_params = {
-            "max_tokens_to_sample": self.max_length,
-            "stop_sequences": ["\n\nHuman:"],
-            "temperature": None,
-            "top_p": None,
-            "top_k": None,
-        }
-        params = self._get_params(inference_kwargs, default_params)
-
-        body = {"prompt": f"\n\nHuman: {prompt}\n\nAssistant:", **params}
+        if self.use_messages_api:
+            default_params: Dict[str, Any] = {
+                "anthropic_version": "bedrock-2023-05-31",
+                "max_tokens": self.max_length,
+                "system": None,
+                "stop_sequences": None,
+                "temperature": None,
+                "top_p": None,
+                "top_k": None,
+            }
+            params = self._get_params(inference_kwargs, default_params)
+
+            body = {"messages": [{"role": "user", "content": prompt}], **params}
+        else:
+            default_params = {
+                "max_tokens_to_sample": self.max_length,
+                "stop_sequences": ["\n\nHuman:"],
+                "temperature": None,
+                "top_p": None,
+                "top_k": None,
+            }
+            params = self._get_params(inference_kwargs, default_params)
+
+            body = {"prompt": f"\n\nHuman: {prompt}\n\nAssistant:", **params}
```

Comment:
I realize this would be more of a feature request, but does Claude support system prompts? In our other generators like `OpenAIGenerator` we support specifying a system prompt at init time. Also for Generators like `Llama2` or `Llama3` we can specify a system prompt by using the appropriate tokens when constructing the prompt template. It feels like Claude is treated a little differently by auto-formatting the prompt without telling the user.

Reply:
When used with messages API you can simply specify `"system"` param as in this test:
https://github.com/deepset-ai/haystack-core-integrations/blob/c16f1877ee3a9b7aa7545d31093662c4980bbbc2/integrations/amazon_bedrock/tests/test_generator.py#L262-L288

Reply:
Ahh okay I see. Sorry, but it's not immediately obvious to me, but can then I just pass a value to `system` as a kwarg to `AmazonBedRockGenerator`?